ts/src/main/scala/ch/epfl/ts/example/BTCArbitrage.scala

```diff
@@ -49,9 +52,11 @@ object BTCArbitrage {
     val bitstampOrderFetcher = builder.createRef(Props(classOf[PullFetchComponent[Order]], bitstampOrderPullFetcher, implicitly[ClassTag[Order]]), "bitstampOrderFetcher")
     // trading agents
     val arbitrageurId = 111L
-    val tradingPriceDelta = 1.0
-    val volume = 50.0
-    val arbitrageur = builder.createRef(Props(classOf[Arbitrageur], 111L, tradingPriceDelta, volume), "arbitrageur")
+    val parameters = new StrategyParameters(
+      Arbitrageur.VOLUME -> NaturalNumberParameter(50),
+      Arbitrageur.PRICE_DELTA -> RealNumberParameter(1.0)
+    )    
+    val arbitrageur = builder.createRef(Props(classOf[Arbitrageur], 111L, parameters), "arbitrageur")
```

Comment:
God I'm lucky to have you looking over my shoulder otherwise I would forget half the design decisions I make!